src/components/Icon/Icon.tsx

```diff
@@ -41,8 +41,23 @@ const getIconSize = (size: IconProps["size"] = "md") => {
   }
 };
 
-const IconRoot = styled.div<Omit<IconProps, "name">>`
-  color: ${(props) => props.color || "inherit"};
+const getColor = (color: IconProps["color"]) => css`
+  color: ${(props) => {
+    switch (color) {
+      case "light":
+        return props.theme.color.white;
+      case "primary":
+        return props.theme.color.primary;
+      case undefined:
+        return "inherit";
+      default:
+        return color;
+    }
+  }};
+`;
+
+const IconRoot = styled.div<Omit<IconProps, "name" | "color">>`
```

Comment:
Farei diventare IconRoot uno `span` in modo che possa essere usato in modo safe dentro ai `p`, `a` e così via